lib/config-generator.js

```diff
@@ -155,67 +117,130 @@ export class ConfigGenerator {
         const extensions = `**/*.{${getExtensions(this.answers)}}`;
 
         let importContent = "import { defineConfig } from \"eslint/config\";\n";
-        const helperContent = `import path from "node:path";
-import { fileURLToPath } from "node:url";
-import { FlatCompat } from "@eslint/eslintrc";
-import js from "@eslint/js";
-
-// mimic CommonJS variables -- not needed if using CommonJS
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = path.dirname(__filename);
-const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: js.configs.recommended});
-`;
         let exportContent = "";
         let needCompatHelper = false;
+        const languages = this.answers.languages ?? ["javascript"];
+        const purpose = this.answers.purpose;
+        const useTs = this.answers.useTs;
 
-        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
-            exportContent += `  { files: ["**/*.js"], languageOptions: { sourceType: "${this.answers.moduleType}" } },\n`;
-        }
-
-        if (this.answers.env?.length > 0) {
-            this.result.devDependencies.push("globals");
-            importContent += "import globals from \"globals\";\n";
-            const envContent = {
-                browser: "globals: globals.browser",
-                node: "globals: globals.node",
-                "browser,node": "globals: {...globals.browser, ...globals.node}"
-            };
-
-            exportContent += `  { files: ["${extensions}"], languageOptions: { ${envContent[this.answers.env.join(",")]} } },\n`;
-        }
-
-        if (this.answers.purpose === "syntax") {
+        if (purpose === "syntax") {
 
             // no need to install any plugin
-        } else if (this.answers.purpose === "problems") {
+        } else if (purpose === "problems") {
             this.result.devDependencies.push("@eslint/js");
             importContent += "import js from \"@eslint/js\";\n";
             exportContent += `  { files: ["${extensions}"], plugins: { js }, extends: ["js/recommended"] },\n`;
         }
```

Comment:
This should probably inside `if (languages.includes("javascript"))`? Currently, it adds a config for `files: ["**/*.{js,mjs,cjs}"]` but then ignores them.